src/ffi/key.rs

```diff
@@ -23,15 +23,27 @@ pub struct AeadParams {
 #[no_mangle]
 pub extern "C" fn askar_key_generate(
     alg: FfiStr<'_>,
+    key_backend: FfiStr<'_>,
     ephemeral: i8,
     out: *mut LocalKeyHandle,
 ) -> ErrorCode {
     catch_err! {
         let alg = alg.as_opt_str().unwrap_or_default();
-        trace!("Generate key: {}", alg);
+        let key_backend = key_backend.as_opt_str().unwrap_or_default();
+        let backend = KeyBackend::from_str(key_backend).unwrap_or_default();
+        trace!("Generate key: {} for {} backend", alg, backend);
         check_useful_c_ptr!(out);
+
         let alg = KeyAlg::from_str(alg)?;
-        let key = LocalKey::generate(alg, ephemeral != 0)?;
+
+        let key = match backend {
+            KeyBackend::Software => LocalKey::generate_with_rng(alg, ephemeral != 0),
+            KeyBackend::SecureElement =>  {
+                let id = Uuid::new_v4().to_string();
+                LocalKey::generate_with_id(alg, &id, ephemeral != 0)
```

Comment:
> Is it appropriate to support 'ephemeral' for these keys?

The underlying library does not yet implement the deletion of keys, so ATM not really no. Once it is there, it would be fine imo to support it. Otherwise we would also have to create a new FFI function for it or make it an optional argument.